pkgs/development/tools/misc/binutils/default.nix

```diff
@@ -247,8 +247,10 @@ stdenv.mkDerivation (finalAttrs: {
     inherit targetPrefix;
     hasGold = enableGold;
     isGNU = true;
-    # Having --enable-plugins is not enough, system has to support dlopen()
-    hasPluginAPI = enableGold && !stdenv.isDarwin && !stdenv.targetPlatform.isWasi;
+    # Having --enable-plugins is not enough, system has to support
+    # dlopen() or equivalent. See config/plugins.m4 and configure.ac
+    # (around PLUGINS) for cases that support or not support plugins.
+    hasPluginAPI = enableGold && !stdenv.isDarwin;
```

Comment:
I think we should drop the `&& !stdenv.isDarwin` here, as it looks like you _could_ opt in to building the plugin API on Darwin (Homebrew does it). We just disable it by default in `enableGold`.

Reply:
Dropped `!stdenv.isDarwin`. Survived the build on `x86_64-darinw` host.